### PR TITLE
Adapt si-id-epu50 to new IDSearch

### PR DESCRIPTION
**Pull request by xresende**
depends on https://github.com/lnls-sirius/dev-packages/pull/1031

### CODE REVIEW

File: si-id-epu50/si_id_epu50/epu.py [hunk 1]
```diff
@@ -259,37 +256,48 @@ def __init__(self, args, callback_update=lambda: 1):
...
+            idname = self.args.pv_prefix
+            self.idparams = _IDSearch.conv_idname_2_parameters(idname)
+            self._pol_state_sel_str = \
+                _IDSearch.conv_idname_2_polarizations(idname)
+            self._pol_state_mon_str = \
+                _IDSearch.conv_idname_2_polarizations_sts(idname)
+            self._pol_none = \
+                self._pol_state_mon_str.index(_IDSearch.POL_NONE_STR)
+            self._pol_undef = \
+                self._pol_state_mon_str.index(_IDSearch.POL_NONE_STR)
```
Comment (anacso17) [inline si-id-epu50/si_id_epu50/epu.py:270]: typo?

File: si-id-epu50/si_id_epu50/epu.py [hunk 2]
```diff
@@ -695,7 +703,7 @@ def gap_set_velocity(self, target: float) -> bool:
             return False
 
     def phase_set(self, target: float) -> bool:
-        if _cte.minimum_phase <= target <= _cte.maximum_phase:
+        if self.idparams.PPARM_MIN <= target <= self.idparams.PPARM_MAX:
```
Comment (anacso17) [inline si-id-epu50/si_id_epu50/epu.py:706]: typos? PPARM -> PPARAM

File: si-id-epu50/si_id_epu50/iocDriver.py [hunk 3]
```diff
@@ -410,7 +429,7 @@ def write(self, reason, value):
 
         # change phase set point
         elif EPUSupport.isPvName(reason, _db.pv_phase_sp):
-            if value >= _cte.minimum_phase and value <= _cte.maximum_phase:
+            if value >= idparams.PPARAM_MIN and value <= idparams.PPARM_MAX:
```
Comment (anacso17) [inline si-id-epu50/si_id_epu50/iocDriver.py:432]: typo PPARM_MAX -> PPARAM_MAX

File: si-id-epu50/si_id_epu50/epu.py [hunk 4]
```diff
@@ -1139,39 +1150,21 @@ def polarization_motion(self, start: bool = False) -> None:
...
-
-        # at this point the configuration must be undefined
-        pol_idx = _cte.polarization_mon.index('undef')
-        self.polarization = pol_idx
+    def update_polarization_status(self) -> int:
+        """ Polarization property. """
+        idname = self.args.pv_prefix
+        pol_idx = _IDSearch.conv_idname_2_polarization_state(
+            idname, pparameter=self.phase, kparameter=self.gap)
+        if pol_idx != self._pol_undef:
```
Comment (anacso17) [inline si-id-epu50/si_id_epu50/epu.py:1165]: why we need this if ?

File: si-id-epu50/si_id_epu50/epu.py [hunk 5]
```diff
@@ -1139,39 +1150,21 @@ def polarization_motion(self, start: bool = False) -> None:
...
+            self.polarization = pol_idx
+        return pol_idx
```
Comment (anacso17) [inline si-id-epu50/si_id_epu50/epu.py:1167]: it seems the return type changed, the function used to return None

File: si-id-epu50/si_id_epu50/epu.py (hunk 4 above)
Comment (xresende) [reply si-id-epu50/si_id_epu50/epu.py:1165]: the way `polarization_motion` is implemented (without check) we need `self.polarization` to be a valid polarization state.

File: si-id-epu50/si_id_epu50/epu.py (hunk 1 above)
Comment (xresende) [reply si-id-epu50/si_id_epu50/epu.py]: fixed

File: si-id-epu50/si_id_epu50/iocDriver.py (hunk 3 above)
Comment (xresende) [reply si-id-epu50/si_id_epu50/iocDriver.py]: fixed

File: si-id-epu50/si_id_epu50/epu.py (hunk 5 above)
Comment (xresende) [reply si-id-epu50/si_id_epu50/epu.py]: yes. we used this int return in a code version we tried in a machine study. but it is not necessary anymore.

File: si-id-epu50/si_id_epu50/epu.py (hunk 4 above)
Comment (xresende) [reply si-id-epu50/si_id_epu50/epu.py]: fixed

File: si-id-epu50/si_id_epu50/epu.py (hunk 5 above)
Comment (xresende) [reply si-id-epu50/si_id_epu50/epu.py]: done

**Review by fernandohds564 — APPROVED** [review]

**Review by murilobalves — APPROVED** [review]

**Review by anacso17 — APPROVED** [review]

**Pull request merged**